tokenserver-auth/src/oauth.rs

```diff
@@ -100,69 +213,28 @@ impl VerifyToken for Verifier {
     /// Verifies an OAuth token. Returns `VerifyOutput` for valid tokens and a `TokenserverError`
     /// for invalid tokens.
     async fn verify(&self, token: String) -> Result<VerifyOutput, TokenserverError> {
```

Comment:
Ditto for myself to test with test vectors here that generate a key pair, generate an access token JWT using the private key, then sets the public key as the `Verifier`'s `jwks` - the tests should cover all of:
- Invalid signatures from all the jwks is an invalid credentials error
- No keys, or improperly formatted token falls back to triggering requests to FxA
   - We trust FxA's word on it if it returns valid claims
   - We also return an invalid credential error if any error is returned from the FxA request
- Expired tokens are an invalid credential error right away
- If we don't have any keys in our environment, we call FxA's `v1/jwks` to get keys
- Test the OK case of verifying the token against the right JWKs
- Make sure that even if the tokens are OK, if the claims don't include the sync scope we also return an incorrect credentials error